000-debug/not-proxied-flag.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Automattic\VIP\NotProxiedFlag;
+
+/**
+ * Display 'Not Proxied' flag when not proxied.
+ *
+ * It can be hard to see if one is really proxied
+ * when accessing wp-admin, especially when sites
+ * are set up with a frontend proxy. Using
+ * such proxies will cause the calling user being
+ * considered not proxied by the backend servers,
+ * even if the caller is using the A8C proxy.
+ *
+ * This mu-plugin will make wp-admin display a small
+ * banner when not proxied and logged in as Automattican,
+ * indicating that the caller is not proxied.
+ */
+
+add_action( 'muplugins_loaded', __NAMESPACE__ . '\enable_not_proxied_flag' );
+
+function enable_not_proxied_flag() {
+	add_action( 'admin_footer', __NAMESPACE__ . '\maybe_show_not_proxied_flag', 9999 ); // output later in the page
+}
+
+function maybe_show_not_proxied_flag() {
+	if ( ! \is_admin() ) {
+		return;
+	}
+
+	if ( ( ! function_exists( '\is_user_logged_in' ) ) || ( ! \is_user_logged_in() ) ) {
+		return;
+	}
+
+	if ( ( ! function_exists( '\is_automattician' ) ) || ( ! \is_automattician() ) ) {
+		return;
+	}
+
+	if ( ( ! function_exists( '\is_proxied_request' ) ) || ( \is_proxied_request() ) ) {
+		return;
+	}
```

Comment:
Here's a much cleaner and easier to read way to write this:

```suggestion
	$is_automattician = function_exists( '\is_automattician' ) && \is_automattician();
	$is_proxied = function_exists( '\is_proxied_request' ) && \is_proxied_request();

	if ( $is_automattician && ! $is_proxied ) {
		return;
	}
```

Right now, the if condition is overly complex and very easy to make an incorrect change (which is bad considering we don't have tests for this).